library/MedEx/API.php

```diff
@@ -1218,92 +1289,31 @@ public function calculateEvents($event, $start_date, $stop_date)
           // and modified some more for MedEx.                                 //
           ///////////////////////////////////////////////////////////////////////
             $data = array();
-            $this->MedEx->logging->log_this("hello in calculateEvents");
-            $this->MedEx->logging->log_this($event['pc_recurrtype']);
-        switch ($event['pc_recurrtype']) {
-            //not recurrent
-            case '0':
-                $data[] = $event;
-                break;
-            case '1':
-            case '3':
-                $event_recurrspec = @unserialize($event['pc_recurrspec'], ['allowed_classes' => false]);
-
-                $rfreq = $event_recurrspec['event_repeat_freq'];
-                $rtype = $event_recurrspec['event_repeat_freq_type'];
-                $exdate = $event_recurrspec['exdate'];
-                list($ny,$nm,$nd) = explode('-', $event['pc_eventDate']);
-                $occurence = $event['pc_eventDate'];
-
-                // prep work to start cooking...
-                // ignore dates less than start_date
-                while (strtotime($occurence) < strtotime($start_date)) {
-                    // if the start date is later than the recur date start
-                    // just go up a unit at a time until we hit start_date
-                    $occurence =& $this->MedEx->events->__increment($nd, $nm, $ny, $rfreq, $rtype);
-                    list($ny,$nm,$nd) = explode('-', $occurence);
-                }
-                //now we are cooking...
-                while ($occurence <= $stop_date) {
-                    $excluded = false;
-                    if (isset($exdate)) {
-                        foreach (explode(",", $exdate) as $exception) {
-                            // occurrence format == yyyy-mm-dd
-                            // exception format == yyyymmdd
-                            if (preg_replace("/-/", "", $occurence) == $exception) {
-                                $excluded = true;
-                            }
-                        }
-                    }
-
-                    if ($excluded == false) {
-                        $data[] = $occurence;
+            switch ($event['pc_recurrtype']) {
+                //not recurrent
+                case '0':
+                    $data[] = $event;
+                    break;
+                case '1':
+                case '3':
+                    $event_recurrspec = @unserialize($event['pc_recurrspec']);
```

Comment:
revert above line change or is a big security vulnerability (let me know if not working with the change)